src/app/doc_api.cpp

```diff
@@ -284,18 +313,19 @@ bool DocApi::cropCel(LayerImage* layer,
             crop_image(newImage.get(),
                        frameBounds.x, frameBounds.y,
                        frameBounds.w, frameBounds.h,
-                       image->pixelFormat() == IMAGE_TILEMAP ?
-                         notile : m_document->bgColor(layer)));
-
-          newCelPos += frameBounds.origin();
+                       bg));
+          if (layer->isTilemap())
+            newCelPos += grid.tileToCanvas(frameBounds.origin());
```

Comment:
This is the final adjust. My recommendation is to debug this when cases where no shrink is needed are resolved.

Reply:
When there's no shrink, this check `if (frameBounds != newImage->bounds())` would fail and the second cropping would not happen.